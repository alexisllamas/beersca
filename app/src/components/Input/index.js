import React from 'react';

const Input = ({ name, labelText, ...props }) => (
  <div className="field">
    <label className="label" htmlFor={name}>
      {labelText}
    </label>
    <div className="control">
      <input id={name} name={name} {...props} />
    </div>
  </div>
);

export default Input;
