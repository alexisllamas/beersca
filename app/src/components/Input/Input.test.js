import React from 'react';
import mount from 'enzyme/mount';
import Input from '.';

describe('Input component', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Input
        labelText="Beer name"
        className="input"
        name="beerName"
        type="text"
        placeholder="Text input"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
