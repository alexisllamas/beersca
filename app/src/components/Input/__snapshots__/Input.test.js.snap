// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly 1`] = `
<Input
  className="input"
  labelText="Beer name"
  name="beerName"
  placeholder="Text input"
  type="text"
>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="beerName"
    >
      Beer name
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        id="beerName"
        name="beerName"
        placeholder="Text input"
        type="text"
      />
    </div>
  </div>
</Input>
`;
